mithril-common/src/chain_reader/fake_chain_reader.rs

```diff
@@ -0,0 +1,82 @@
+use async_trait::async_trait;
+use std::{collections::VecDeque, sync::Mutex};
+
+use crate::{entities::ChainPoint, StdResult};
+
+use super::{ChainBlockNextAction, ChainBlockReader};
+
+/// [FakeChainReader] is a fake implementation of [ChainBlockReader] for testing purposes.
+pub struct FakeChainReader {
+    chain_point_next_actions: Mutex<VecDeque<ChainBlockNextAction>>,
+}
+
+impl FakeChainReader {
+    /// Creates a new [FakeChainReader] instance.
+    pub fn new(chain_point_next_actions: Vec<ChainBlockNextAction>) -> Self {
+        Self {
+            chain_point_next_actions: Mutex::new(chain_point_next_actions.into()),
+        }
+    }
+}
+
+#[async_trait]
+impl ChainBlockReader for FakeChainReader {
+    async fn set_chain_point(&mut self, _point: &ChainPoint) -> StdResult<()> {
+        Ok(())
+    }
+
+    async fn get_next_chain_block(&mut self) -> StdResult<Option<ChainBlockNextAction>> {
+        Ok(self.chain_point_next_actions.lock().unwrap().pop_front())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::cardano_block_scanner::ScannedBlock;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_get_next_chain_block() {
+        let chain_point = ChainPoint {
+            slot_number: 1,
+            block_number: 1,
+            block_hash: "point-hash-1".to_string(),
+        };
+        let expected_chain_point_next_actions = vec![
+            ChainBlockNextAction::RollForward {
+                next_point: ChainPoint {
+                    slot_number: 1,
+                    block_number: 1,
+                    block_hash: "point-hash-1".to_string(),
+                },
+                parsed_block: ScannedBlock::new("hash-1", 1, 10, 20, Vec::<&str>::new()),
+            },
+            ChainBlockNextAction::RollForward {
+                next_point: ChainPoint {
+                    slot_number: 2,
+                    block_number: 2,
+                    block_hash: "point-hash-2".to_string(),
+                },
+                parsed_block: ScannedBlock::new("hash-2", 2, 11, 21, Vec::<&str>::new()),
+            },
+            ChainBlockNextAction::RollBackward {
+                rollback_point: ChainPoint {
+                    slot_number: 1,
+                    block_number: 1,
+                    block_hash: "point-hash-1".to_string(),
+                },
+            },
+        ];
+        let mut chain_reader = FakeChainReader::new(expected_chain_point_next_actions.clone());
+        chain_reader.set_chain_point(&chain_point).await.unwrap();
```

Comment:
Is it useful to call this method ? It does nothing on FakeChainReader.
If it's not, we can also remove the `chain_point`variable
If we just want to check that this function can be called without crash, we may create a dedicated test for that

Reply:
Done